src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveDefinitionTemplatesPassTest.php

```diff
@@ -139,6 +139,25 @@ public function testProcessDoesNotCopyDecoratedService()
         $this->assertNull($def->getDecoratedService());
     }
 
+    public function testProcessDoesNotDropShared()
+    {
+        $container = new ContainerBuilder();
+
+        $container
+            ->register('parent')
+        ;
+
+        $container
+            ->setDefinition('child', new DefinitionDecorator('parent'))
+            ->setShared(false)
```

Comment:
@dosten the shared flag is always taken from the child (same than the scope in 2.x, as this is the replacement for the scope)